crates/fluvio-sc/src/services/auth/mod.rs

```diff
@@ -96,4 +96,50 @@ mod common {
             Self { global_ctx, auth }
         }
     }
+
+    /// Authorization that allows only read only ops and any ops on spu objects
+    #[derive(Debug, Clone)]
+    pub struct ReadOnlyAuthorization {}
+
+    #[async_trait]
+    impl Authorization for ReadOnlyAuthorization {
+        type Context = ReadOnlyAuthContext;
+
+        async fn create_auth_context(
+            &self,
+            _socket: &mut FluvioSocket,
+        ) -> Result<Self::Context, AuthError> {
+            Ok(ReadOnlyAuthContext {})
+        }
+    }
+
+    impl ReadOnlyAuthorization {
+        pub fn new() -> Self {
+            Self {}
+        }
+    }
+
+    #[derive(Debug)]
+    pub struct ReadOnlyAuthContext {}
+
+    #[async_trait]
+    impl AuthContext for ReadOnlyAuthContext {
```

Comment:
Can you add a unit test for this?

Reply:
added unit test and also added no-k8 behavior